src/ui/file_select_sidebar.tsx

```diff
@@ -34,62 +37,62 @@ function FileSelectSideBar(props){
         "" :
         fileList.length + " File" + (fileList.length > 1 ? "s" : "") + " Found:";
 
-    const FileSelectButtons = ({primary}) => <div>
+    const FileSelectButtons = ({ primary }) => <div>
         <Button onClick={() => onButtonClick(inputFile)} className={"m-1"}
-                variant={primary ? "primary" : "outline-light"}>
+            variant={primary ? "primary" : "outline-light"}>
             Select {primary ? "" : "New"} Gerber Files
         </Button>
 
         <Button onClick={() => onButtonClick(inputFile2)} className={"m-1"}
-                variant={primary ? "primary" : "outline-light"}>
+            variant={primary ? "primary" : "outline-light"}>
             Select {primary ? "" : "New"} Folder
         </Button>
 
         <Button onClick={() => onButtonClick(inputFile3)} className={"m-1"}
-                variant={primary ? "primary" : "outline-light"}>
+            variant={primary ? "primary" : "outline-light"}>
             Select {primary ? "" : "New"} ZIP
         </Button>
     </div>;
 
     return <div className={"col-4 dark-bg p-3 h-100"}>
         <input type='file' id='files' ref={inputFile} className={"hidden"}
-               multiple="multiple" accept={".g*,.zip,.txt,.xln,.drl"}
-               onInput={() => {
-                   if (inputFile.current.files.length > 0) {
-                       setRootFileName("")
-                       setFileList(inputFile.current.files)
-                   }
-               }}/>
-        <input type='file' id='folder' ref={inputFile2} className={"hidden"} directory=""
-               webkitdirectory="" onInput={() => {
-                   if (inputFile2.current.files.length > 0) {
-                       setRootFileName(inputFile2.current.files[0].webkitRelativePath.split("/")[0]);
-                       setFileList(inputFile2.current.files);
-                   }
-               }}/>
+            multiple={true} accept={".g*,.zip,.txt,.xln,.drl"}
+            onInput={() => {
+                if (inputFile.current?.files != null && inputFile.current.files.length > 0) {
+                    setRootFileName("")
+                    setFileList(Array.from(inputFile.current.files))
+                }
+            }} />
+        <input type='file' id='folder' ref={inputFile2} className={"hidden"}
+            onInput={() => {
```

Comment:
With these changes we don't seem to be able to upload entire directories anymore. Any particular reason to remove the `directory` and `webkitdirectory` attributes here?

Reply:
Well spotted. It requires some tricking to add the attributes, that's why I did not bother. But they're absolutely required for directory uploads. Fixed.